openapi/generator.go

```diff
@@ -1161,3 +1177,21 @@ func fieldNameFromTag(sf reflect.StructField, tagName string) string {
 	}
 	return name
 }
+
+/// parseExampleValue is used to transform the string representation of the example value to the correct type.
+func parseExampleValue(rType reflect.Type, value string) (interface{}, error) {
+	switch rType.Kind() {
+	case reflect.Bool:
+		return strconv.ParseBool(value)
+	case reflect.String:
+		return value, nil
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return strconv.ParseInt(value, 10, rType.Bits())
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return strconv.ParseUint(value, 10, rType.Bits())
+	case reflect.Float32, reflect.Float64:
+		return strconv.ParseFloat(value, rType.Bits())
+	default:
+		return nil, errors.New("parseExampleValue does not support type: " + rType.String())
```

Comment:
I'd remove the function name in the error, it doesn't bring any value. Return smthing like `unsupported type: %s`with `fmt.Errorf`, and eventually in the parent give more context about the error, like "couldn't parse example value of field...".

Reply:
Changed. Also reworded the message in parent a bit to give the field name as well.